Allow disabling video autoplay

timomyllymaki opened the pull request:
Hi,

Fixes #393.

Adds an option to disable video autoplay.

Kind of implemented as suggested here https://github.com/sindresorhus/caprine/issues/393#issuecomment-452622199. It doesn't actually capture the first frame of the video but instead takes one from the video parent element background-image which holds a snapshot of the video. Its not the first frame but I didn't think it was that important as this was really easy to implement.

timomyllymaki commented:
Yes sure. I'll try to fix them asap.

timomyllymaki commented:
I fixed alI the requested changes. Also changed the mutation observers a little. Seems like it's a lot better to not iterate over mutations but to search for the videos by tag when we have mutations. By my tests this way seems to perform better.

spramij commented:
@timomyllymaki This needs a rebase from `origin:master` to accommodate for changes in code that were done since.

sindresorhus commented:
When this feature is enabled, I noticed that the videos lose their round corners. Can you handle that?

sindresorhus commented on source/config.ts:
```diff
@@ -36,7 +36,8 @@ const defaults = {
 	quitOnWindowClose: false,
-	keepMeSignedIn: true
+	keepMeSignedIn: true,
+	videoAutoplay: true
```
`autoplayVideos`

sindresorhus commented on source/menu.ts:
```diff
@@ -173,6 +173,16 @@ Press Command/Ctrl+R in Caprine to see your changes.
 			}
 		},
+		{
+			label: 'Video Autoplay',
```
`Autoplay Videos`

sindresorhus commented:
Can you move as much of the feature as possible into a separate file? For maintenance reasons.

timomyllymaki commented:
I refactored the feature to own .ts and .css -files. Also fixed the rounded corners (thus new .css -file) and changed videoAutoplay -> autoplayVideos. Let me know if there's something else to fix!

CvX commented:
Almost there! 🙂 The preview has only top corners rounded, as opposed to all four corners rounded with autoplay enabled:

<img src="https://user-images.githubusercontent.com/66961/58809809-eaeb3480-861c-11e9-86e3-31fc7c50a09c.png" width="320" alt="autoplay disabled" > <img src="https://user-images.githubusercontent.com/66961/58809808-eaeb3480-861c-11e9-95e7-764a95886504.png" width="320" alt="autoplay enabled" >

timomyllymaki commented:
Oh, I had only tested it with one of these videos: 

![image](https://user-images.githubusercontent.com/10549023/58813594-c39c6380-862c-11e9-9998-1377f68594c7.png)

Made a fix which hopefully solves the problem!

CvX commented:
> Oh, I had only tested it with one of these videos:

Ha, I didn't even think about videos with a description below! 😃
It's great now! 🙂 

<img src="https://user-images.githubusercontent.com/66961/58816510-f729be80-8629-11e9-9271-5786cc93804b.png" width="320" alt="autoplay disabled, after changes">

sindresorhus commented:
Thank you!

sindresorhus commented:
@timomyllymaki You need to submit the PR URL to https://issuehunt.io/r/sindresorhus/caprine/issues/393 to claim the bounty.
